### PR TITLE
commands, jobs: add go-build job

### PR DESCRIPTION
**Pull request by kopiczko**
- Example use (devctl): https://github.com/giantswarm/devctl/pull/24
- Exemple build (devctl): https://circleci.com/gh/giantswarm/devctl/125?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

### CODE REVIEW

File: src/commands/go-build.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    type: "string"
+steps:
+  - restore_cache:
+      keys:
+        - go-build-go-mod-v1-{{ checksum "go.sum" }}
+  - run: |
+      architect project version > .project_version
+  - run: |
+      cat .project_version
+  - run: |
```
Comment (MarcelMue) [inline src/commands/go-build.yaml]: Why do all of these need to be different `run` steps? Can they not be done in one step?

Comment (kopiczko) [reply src/commands/go-build.yaml]: I was talking about this during orb walkthrough. This is so we have output of each command in a different block and know which of them actually failed. See example build: https://circleci.com/gh/giantswarm/devctl/125?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Comment (MarcelMue) [reply src/commands/go-build.yaml]: I get it and it makes sense, the code still makes my eyes bleed a bit. Not blocking.

Comment (kopiczko) [reply src/commands/go-build.yaml]: This is ~10 bash commands. I would argue you can get trough 10 lines with a bit bleeding eyes :) IMO this is way simpler to both implement and understand than the current architect approach which is running most of those command under the hood anyway.

**Pull request merged**